From 2fdaf41cf9d8cb29d87c5d191809dfa616cec518 Mon Sep 17 00:00:00 2001
From: rc420head <dennislebhardt79@gmail.com>
Date: Tue, 12 Nov 2019 21:47:25 -0600
Subject: [PATCH 07/19] fix error com.google.android.gms.persistent

11-10 17:26:32.914  2574  2574 E GmsApplication: /system/etc/sysconfig/google.xml must have <allow-in-power-save package="com.google.android.gms"/>.
11-10 17:26:32.919  2574  2574 E GmsApplication: Unable to re-add to doze whitelist
11-10 17:26:32.919  2574  2574 E GmsApplication: java.lang.SecurityException: Calling app u0a106 is not on whitelist
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.os.Parcel.createException(Parcel.java:2071)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.os.Parcel.readException(Parcel.java:2039)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.os.Parcel.readException(Parcel.java:1987)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.usage.IUsageStatsManager$Stub$Proxy.whitelistAppTemporarily(IUsageStatsManager.java:995)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.usage.UsageStatsManager.whitelistAppTemporarily(UsageStatsManager.java:1036)
11-10 17:26:32.919  2574  2574 E GmsApplication: at rrc.startService(:com.google.android.gms@19420046@19.4.20 (120400-271418971):5)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.content.ContextWrapper.startService(ContextWrapper.java:669)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.content.ContextWrapper.startService(ContextWrapper.java:669)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.content.ContextWrapper.startService(ContextWrapper.java:669)
11-10 17:26:32.919  2574  2574 E GmsApplication: at dzh.a(:com.google.android.gms@19420046@19.4.20 (120400-271418971):11)
11-10 17:26:32.919  2574  2574 E GmsApplication: at dzh.onReceive(:com.google.android.gms@19420046@19.4.20 (120400-271418971):7)
11-10 17:26:32.919  2574  2574 E GmsApplication: at qlz.onReceive(:com.google.android.gms@19420046@19.4.20 (120400-271418971):8)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread.handleReceiver(ActivityThread.java:3788)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread.access$1400(ActivityThread.java:219)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1870)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.os.Handler.dispatchMessage(Handler.java:107)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.os.Looper.loop(Looper.java:214)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread.main(ActivityThread.java:7356)
11-10 17:26:32.919  2574  2574 E GmsApplication: at java.lang.reflect.Method.invoke(Native Method)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
11-10 17:26:32.919  2574  2574 E GmsApplication: Caused by: android.os.RemoteException: Remote stack trace:
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController.addPowerSaveTempWhitelistAppDirectInternal(DeviceIdleController.java:2378)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController.addPowerSaveTempWhitelistAppInternal(DeviceIdleController.java:2360)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController.addPowerSaveTempWhitelistAppChecked(DeviceIdleController.java:2324)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController$BinderService.addPowerSaveTempWhitelistApp(DeviceIdleController.java:1561)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.usage.UsageStatsService$BinderService.whitelistAppTemporarily(UsageStatsService.java:1334)va:1987)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.usage.IUsageStatsManager$Stub$Proxy.whitelistAppTemporarily(IUsageStatsManager.java:995)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.usage.UsageStatsManager.whitelistAppTemporarily(UsageStatsManager.java:1036)
11-10 17:26:32.919  2574  2574 E GmsApplication: at rrc.startService(:com.google.android.gms@19420046@19.4.20 (120400-271418971):5)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.content.ContextWrapper.startService(ContextWrapper.java:669)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.content.ContextWrapper.startService(ContextWrapper.java:669)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.content.ContextWrapper.startService(ContextWrapper.java:669)
11-10 17:26:32.919  2574  2574 E GmsApplication: at dzh.a(:com.google.android.gms@19420046@19.4.20 (120400-271418971):11)
11-10 17:26:32.919  2574  2574 E GmsApplication: at dzh.onReceive(:com.google.android.gms@19420046@19.4.20 (120400-271418971):7)
11-10 17:26:32.919  2574  2574 E GmsApplication: at qlz.onReceive(:com.google.android.gms@19420046@19.4.20 (120400-271418971):8)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread.handleReceiver(ActivityThread.java:3788)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread.access$1400(ActivityThread.java:219)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1870)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.os.Handler.dispatchMessage(Handler.java:107)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.os.Looper.loop(Looper.java:214)
11-10 17:26:32.919  2574  2574 E GmsApplication: at android.app.ActivityThread.main(ActivityThread.java:7356)
11-10 17:26:32.919  2574  2574 E GmsApplication: at java.lang.reflect.Method.invoke(Native Method)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
11-10 17:26:32.919  2574  2574 E GmsApplication: Caused by: android.os.RemoteException: Remote stack trace:
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController.addPowerSaveTempWhitelistAppDirectInternal(DeviceIdleController.java:2378)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController.addPowerSaveTempWhitelistAppInternal(DeviceIdleController.java:2360)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController.addPowerSaveTempWhitelistAppChecked(DeviceIdleController.java:2324)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.DeviceIdleController$BinderService.addPowerSaveTempWhitelistApp(DeviceIdleController.java:1561)
11-10 17:26:32.919  2574  2574 E GmsApplication: at com.android.server.usage.UsageStatsService$BinderService.whitelistAppTemporarily(UsageStatsService.java:1334)

Change-Id: I114fd5ef4ea4d72120b4fd73385f808ae2d7f3a5
---
 data/etc/platform.xml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/data/etc/platform.xml b/data/etc/platform.xml
index 6897c01844..52127ddb7d 100644
--- a/data/etc/platform.xml
+++ b/data/etc/platform.xml
@@ -333,4 +333,6 @@
 
     <!-- Allow IMS service entitlement app to schedule jobs to run when app in background. -->
     <allow-in-power-save-except-idle package="com.android.imsserviceentitlement" />
+
+    <allow-in-power-save package="com.google.android.gms"/>
 </permissions>
-- 
2.34.1

